third_party/terraform/resources/resource_compute_router_nat.go

```diff
@@ -302,8 +279,105 @@ func resourceComputeRouterNatRead(d *schema.ResourceData, meta interface{}) erro
 	return nil
 }
 
-func resourceComputeRouterNatDelete(d *schema.ResourceData, meta interface{}) error {
+func resourceComputeRouterNatUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
 
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	routerName := d.Get("router").(string)
+	natName := d.Get("name").(string)
+
+	routerLock := getRouterLockName(region, routerName)
+	mutexKV.Lock(routerLock)
+	defer mutexKV.Unlock(routerLock)
+
+	// Get router
+	router, err := config.clientComputeBeta.Routers.Get(project, region, routerName).Do()
+	if err != nil {
+		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
+			return fmt.Errorf("Router %s not found", routerName)
+		}
+
+		return fmt.Errorf("Error reading parent router %s: %s", routerName, err)
+	}
+
+	var nat *computeBeta.RouterNat
+	nats := router.Nats
+	for _, nat = range nats {
+		if nat.Name == natName {
+			break
+		}
+	}
+	nat.ForceSendFields = []string{
+		"natIps",
+		"subnetworks",
+		"minPortsPerVm",
+		"udpIdleTimeoutSec",
+		"icmpIdleTimeoutSec",
+		"tcpEstablishedIdleTimeoutSec",
+		"tcpTransitoryIdleTimeoutSec",
+	}
+
+	nat.MinPortsPerVm = int64(d.Get("min_ports_per_vm").(int))
+	nat.UdpIdleTimeoutSec = int64(d.Get("udp_idle_timeout_sec").(int))
+	nat.IcmpIdleTimeoutSec = int64(d.Get("icmp_idle_timeout_sec").(int))
+	nat.TcpEstablishedIdleTimeoutSec = int64(d.Get("tcp_established_idle_timeout_sec").(int))
+	nat.TcpTransitoryIdleTimeoutSec = int64(d.Get("tcp_transitory_idle_timeout_sec").(int))
+
+	if v, ok := d.GetOk("nat_ip_allocate_option"); ok {
+		nat.NatIpAllocateOption = v.(string)
+	}
+
+	if v, ok := d.GetOk("nat_ips"); ok {
+		nat.NatIps = convertStringArr(v.(*schema.Set).List())
+	} else {
+		nat.NatIps = []string{}
+	}
+
+	if v, ok := d.GetOk("subnetwork"); ok {
+		nat.Subnetworks = expandSubnetworks(v.(*schema.Set).List())
+	} else {
+		nat.Subnetworks = []*computeBeta.RouterNatSubnetworkToNat{}
+	}
+
+	if v, ok := d.GetOk("source_subnetwork_ip_ranges_to_nat"); ok {
+		nat.SourceSubnetworkIpRangesToNat = v.(string)
+	}
+
+	if v, ok := d.GetOk("log_config"); ok {
+		nat.LogConfig = expandLogConfig(v)
+	}
+
+	log.Printf("[INFO] Updating nat %s: +%v", natName, nat)
+	obj := &computeBeta.Router{
+		Nats: nats,
+	}
+
+	log.Printf("[DEBUG] Updating router %s/%s with nats: %+v", region, routerName, nats)
+	op, err := config.clientComputeBeta.Routers.Patch(project, region, router.Name, obj).Do()
+	if err != nil {
+		return fmt.Errorf("Error patching router %s/%s: %s", region, routerName, err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s/%s", project, region, routerName, natName))
```

Comment:
```suggestion
```

Reply:
This is just redundant, right?

Reply:
yeah, it was there before so I was like fewer changes I make the better, but I can remove it